runtime/src/accounts_index.rs

```diff
@@ -842,47 +854,55 @@ impl<T: 'static + Clone + IsCached + ZeroLamport> AccountsIndex<T> {
             .map(WriteAccountMapEntry::from_account_map_entry)
     }
 
-    fn new_entry() -> AccountMapEntry<T> {
-        Arc::new(AccountMapEntryInner {
-            ref_count: AtomicU64::new(0),
-            slot_list: RwLock::new(SlotList::with_capacity(1)),
-        })
-    }
-
-    fn insert_new_entry_if_missing(&self, pubkey: &Pubkey) -> (WriteAccountMapEntry<T>, bool) {
-        let new_entry = Self::new_entry();
-        let mut w_account_maps = self.get_account_maps_write_lock();
-        self.insert_new_entry_if_missing_with_lock(pubkey, &mut w_account_maps, new_entry)
+    fn insert_new_entry_if_missing(
+        &self,
+        pubkey: &Pubkey,
+        slot: Slot,
+        info: &T,
+        lock: Option<&mut ReadWriteLockMapType<T>>,
```

Comment:
maybe a better name than `lock` here. The name and type don't help understand what this is.

Reply:
updated to:
``` 
   fn insert_new_entry_if_missing(
        &self,
        pubkey: &Pubkey,
        slot: Slot,
        info: &T,
        w_account_maps: Option<&mut AccountMapsWriteLock<T>>,
    ) -> Option<WriteAccountMapEntry<T>> {
```